src/Core/src/Credentials/ConfigurationProvider.php

```diff
@@ -13,20 +17,72 @@
  */
 final class ConfigurationProvider implements CredentialProvider
 {
+    private $logger;
+
+    private $httpClient;
+
+    public function __construct(?LoggerInterface $logger = null, ?HttpClientInterface $httpClient = null)
```

Comment:
Let's flip these arguments to be constant with other classes. 

```suggestion
    public function __construct(?HttpClientInterface $httpClient = null, ?LoggerInterface $logger = null)
```